mapper/software_management/README.md

```diff
@@ -0,0 +1,126 @@
+## Components
+
+### Thin-Edge-Json
+
+Responsibilities:
+* Define the API between a cloud provider and the DM-agent
+* Ability to convey as well:
+   * desired software lists
+   * software operation lists
+   * device profile
+   * current software lists
+   * operation status
+* Link a response to a request
+
+Out of scope:
+* Send the operation logs to the cloud.
+
+Open questions:
+* How to shorten the current software list sent by a device?
+
+### Specific mapper - e.g. C8y Mapper
+
+Responsibilities:
+* Register the device, so it will receive the software management requests.
+* Send the list of already installed software packages as all future update operations are derived from this initial list.
+* Listen and translate the operations received from the cloud.
+* Forward these operations to the DM-agent and listen for the responses.
+* Translate the DM-Agent responses and forward them to the cloud.
+
+Open questions:
+* Each software update message has an id, to be used for the response ...
+   * Do we have to report the result in one go ... or can we have several partial responses using the same id?
+* Cumulocity doesn't let you send a response to a specific installation request based on an id.
```

Comment:
This is yes and no. **SmartREST2.0** doesn't let you send a response based on an ID. This is a feature of SmartREST2.0. If we go to custom template of SmartREST, we can get an operation ID from cloud together with operation's contents.

Reply:
Which of SmartREST and SmartREST2.0 supports the IDs?